src/datahike/index/persistent_set.cljc

```diff
@@ -166,6 +166,7 @@
     (if-let [cached (wrapped/lookup cache address)]
       cached
       (let [node (k/get store address nil {:sync? true})]
+        (assert (not (nil? node)) "Node not found in storage.")
```

Comment:
This should never happen. All the nodes we refer to in our addresses/pointers must always exist, otherwise the store got corrupted or a write operation was inconsistent (e.g. the underlying store did not ensure its durability or mixed up the causal order of events [I think Redis, does, but it does not immediately store durably, so something like this could go wrong in a faulty store backend]). If your comment is about the assertion, yes I will change that to an Exception now.

But we also could allow it and retry. In this case you would be able to sync the store from another system out of order (e.g. through datsync, rsync or file copying or whatever) and your operations would just stall instead of break. But I think it is better to just copy everything except for the roots first and overwrite the roots last in an atomic operation (in the same order commit writes), that way you can sync without interrupting operations at all.